src/components/Button/Button.tsx

```diff
@@ -122,17 +123,37 @@ export const Button: FC<ButtonProps> = React.forwardRef(
       { [styles.buttonSecondary]: mergedVariant === ButtonVariant.Secondary },
       { [styles.buttonSystemUi]: mergedVariant === ButtonVariant.SystemUI },
       {
-        [styles.buttonDisruptive]:
+        [styles.buttonSystemUiWarning]:
+          warning && mergedVariant === ButtonVariant.SystemUI,
+      },
+      {
+        [styles.buttonSystemUiDisruptive]:
+          disruptive && mergedVariant === ButtonVariant.SystemUI,
+      },
+      {
+        [styles.buttonDefaultWarning]:
+          warning && mergedVariant === ButtonVariant.Default,
+      },
+      {
+        [styles.buttonDefaultDisruptive]:
```

Comment:
Add UTs snapshotting the warning state for each button type.

Reply:
Added in UTs 👍